opm/simulators/flow/LogOutputHelper.cpp

```diff
@@ -499,100 +442,75 @@ production(const std::size_t reportStepNum) const
-            tmp_values[8] = 0.0;
+        values[Ix::WatGasRatio] = values[Ix::WaterRate] / values[Ix::GasRate];
+        if (std::isnan(values[Ix::WatGasRatio])) {
+            values[Ix::WatGasRatio] = 0.0;
         }
 
-        this->outputProductionReportRecord_(tmp_values, tmp_names);
+        this->outputProductionReportRecord_(values, names);
     }
 
-    for (const auto& wname : schedule_.wellNames(reportStepNum)) {
-        const auto& well = schedule_.getWell(wname, reportStepNum);
+    for (const auto& wname : this->schedule_.wellNames(reportStepNum)) {
+        const auto& well = this->schedule_.getWell(wname, reportStepNum);
 
-        // Ignore injector wells
+        // Ignore injection wells for the production report.
         if (well.isInjector()) {
             continue;
         }
 
-        tmp_names[0] = wname; // WellProdDataType::WellName
-
-
-        auto wName = static_cast<std::string>(wname);
-        auto get = [&st, &wName](const std::string& vector)
-        {
-            const auto key = vector + ':' + wName;
-            return st.has(key) ? st.get(key) : 0.0;
-        };
-
-        const auto& controls = well.productionControls(st);
-        using CMode = Well::ProducerCMode;
-
-        auto fctl = [](const auto wmctl) -> std::string
-        {
-            switch (wmctl) {
-            case CMode::ORAT: return "ORAT";
-            case CMode::WRAT: return "WRAT";
-            case CMode::GRAT: return "GRAT";
-            case CMode::LRAT: return "LRAT";
-            case CMode::RESV: return "RESV";
-            case CMode::THP:  return "THP";
-            case CMode::BHP:  return "BHP";
-            case CMode::CRAT: return "CRate";
-            case CMode::GRUP: return "GRUP";
-            default: return "none";
-            }
-        };
-
-        tmp_names[1] = fctl(controls.cmode); // WellProdDataType::CTRLMode
-
-        tmp_values[0] = well.getHeadI() + 1; // WellProdDataType::WellLocationi
-        tmp_values[1] = well.getHeadJ() + 1; // WellProdDataType::WellLocationj
-        tmp_values[2] = get("WOPR"); // WellProdDataType::OilRate
-        tmp_values[3] = get("WWPR"); // WellProdDataType::WaterRate
-        tmp_values[4] = get("WGPR"); // WellProdDataType::GasRate
-        tmp_values[5] = get("WVPR"); // WellProdDataType::FluidResVol
-        tmp_values[6] = get("WWCT"); // WellProdDataType::WaterCut
-        tmp_values[7] = get("WGOR"); // WellProdDataType::GasOilRatio
-        tmp_values[9] = get("WBHP"); // WellProdDataType::BHP
-        tmp_values[10] = get("WTHP"); // WellProdDataType::THP
-        //tmp_values[11] = 0; //WellProdDataType::SteadyStatePI //
-
-        tmp_values[8] = tmp_values[3] / tmp_values[4]; // WellProdDataType::WaterGasRatio
-        if (std::isnan(tmp_values[8])) {
-            tmp_values[8] = 0.0;
+        std::vector<Scalar> values(WellProdDataType::numWPValues, 0.0);
+        std::vector<std::string> names(WellProdDataType::numWPNames, "");
```

Comment:
default ctor

Reply:
> default ctor

Agreed.  I've pushed an update to that effect.